CPS-XXXX/README.md

```diff
@@ -0,0 +1,76 @@
+
+## Problem
+The [Ouroboros Praos paper](<https://eprint.iacr.org/2017/573.pdf>) states that it is the "*probability of being permitted to issue a block*" that "is proportional to the relative stake a player has in the system".  However, it is clear from the security analysis section that it is the _probability of contributing to the ledger_ that is supposed to be proportional to the relative stake of the participant.  This is an important distinction because it matters little if a player has stake weighted permission to make blocks if the protocol later disproportionately drops that players blocks despite their honest performance.
+
+With the current Ouroboros implementation where slots have 1 second duration, it is not uncommon to have consecutive slot leaders that are only 1 second apart.  Remote participants may be unable to receive the previous block within 1 second and so these block producers will make blocks that result in forks.  Ouroboros settles such forks by preferring the fork whose terminal block has the lowest VRF value.  This seems like a fair method because it is deterministic, neither player can alter the outcome, and each player has an equal chance of winning.  However, the problem is that the more geographically decentralised players will face a disproportionately greater number of "fork battles".  This in turn means that they will get more of their blocks dropped.  The net result is that more geographically decentralised participants do not receive stake weighted _probability of contributing to the ledger_.  This is not only unfair, but also represents a deviation from the Ouroboros security analysis assumptions.
+
+This might seem like a minor problem, but the effect is significant.  If the majority of the network reside in USA - Europe with close connectivity and less than 1 second propagation delays, then those participant pools will see 5% of their blocks suffering "fork battles" which will only occur when another pool is awarded the exact same slot (ie: a "slot battle").  They will lose half of these battles on average causing 2.5% of their blocks to get dropped, or "orphaned".
+
+However, for a pool that happens to reside on the other side of the world where network delays might be just over 1 second, this pool will suffer "fork battles" not only with pools awarded the same slot, but also the slot before, and the slot after.  In other words, this geographically decentralised pool will suffer 3 times the number of slot battles, amounting to 15% of its blocks, and resulting in 7.5% of its blocks getting dropped.  The numbers are even worse for a pool suffering 2 second network delays because it will suffer 5 times the number of "fork battles" and see 12.5% of its blocks "orphaned".  This not only results in an unfair reduction in rewards, but also the same magnitude reduction in contribution to the ledger.
```

Comment:
This CPS does not provide data showing that 1s is not enough for blocks to propagate anywhere in the world with required hardware, connection and configuration. Without such data, it's impossible to determine if it's needed or not, and what solution would work.

Reply:
I often get 1000ms ping times just crossing India's border: though AWS Mumbai is probably exempt from such delays, which I imagine result from "great firewall" type packet inspection from the newly founded & somewhat ill-equipped surveillance state here.

Reply:
This block was produced by my BP earlier today. It was full at 86.57kB in size, containing 64 transactions, and 66.17kB of scripts: https://cexplorer.io/block/c740f9ce8b25410ddb938ff8c42e12738c18b7fd040ae5224c53fb45f04b3ba0

These are the delays (from beginning of the slot) before each of _my own relays_ included this block in their chains:

- Relay 1 (ARM on same LAN) → Delayed=0.263s
- Relay 2 (AMD on adjacent LAN) → Delayed=0.243s
- Relay 3 (ARM approx 5 Km away) → Delayed=0.288s
- Relay 4 (AMD Contabo vps in USA) → Delayed=2.682s
- Relay 5 (ARM Netcup vps in USA) → Delayed=1.523s

The average propagation delay by nodes pinging the data to pooltool was 1.67 seconds: https://pooltool.io/realtime/11169975

Reply:
@TerminadaDrep Could you add the above delay metrics to the CPS? I think having empirical data would help strengthen the case of this CPS. Also, could you indicate whether your BP is locally controlled or in a vps? I'm guessing it is locally controlled.

Reply:
Here's a visualisation of the block mentioned by terminada before produced by his TERM BP. I assume located in Australia. 
 
![Screenshot_2024-12-03-13-16-57-65_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/1a910e10-4e02-440b-a276-4b75a706c2bc)

It's all mobile screenshots only, as I'm out of office this week.

But I tried to show the propagation experience as reported from many nodes around the globe (above screenshot shows only half of them). You can see almost all nodes take already a second to only hear about the existence of the new block, by receiving the header announcement. 
Only then the fetch request and body download starts.
And only as the last (blue) step the local validation happens.

What needs to be noted here: I sorted these reporting nodes by continent. First row shows a relay node from Sidney Australia. And this node already has a long waiting time until he heard about the header. 

This is also the case for the Asian nodes at the end of the list 

![Screenshot_2024-12-03-13-21-18-59_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/7b3f2bba-1317-4e23-b378-641b10f76e9c)

Reply:
It is interesting that the delay is clearly in the Australian part of the internet.  Perhaps the Aussie national broadband network (NBN) was more congested than usual at this time.

Certainly this block had worse than usual propagation delays.

Reply:
>  So I also disagree with one of the points in the CPS, in expectations of future transmission and latency improvements.

@gufmar I'm not an expert on networks, but I really don't think we should be relying on improvements to network throughput here due to the [rebound effect](https://en.wikipedia.org/wiki/Rebound_effect_(conservation)). Demand will very likely increase to use up any extra "slack" (eg, Leios, block size increases, and even non-blockchain demand). If network capacity doubles but so does demand, we can easily find ourselves here again.

Reply:
Yes sorry if I was unclear. I wanted to say I disagree with the argument made at https://github.com/cardano-foundation/CIPs/blob/e7bf9b4c103f3841f2d8364e78905c1183ee9526/CPS-XXXX/README.md#arguments-against-correcting-this-unfairness

because I don't expect we will see significant improvements in network latency which is the main limiting factor here with the TCP ACK back and forth packets, not so much the Throughput.